lib/trino-parquet/src/main/java/io/trino/parquet/writer/valuewriter/FixedLenByteArrayShortDecimalValueWriter.java

```diff
@@ -53,11 +53,59 @@ private byte[] paddingLong(long unscaledValue)
     {
         int numBytes = getTypeLength();
         byte[] result = new byte[numBytes];
-        int shift = 8 * (numBytes - 1);
-
-        for (int i = 0; i < numBytes; i++) {
-            result[i] = (byte) (unscaledValue >> shift);
-            shift -= 8;
+        switch (numBytes) {
+            case 1:
+                result[0] = (byte) unscaledValue;
+                break;
+            case 2:
+                result[0] = (byte) (unscaledValue >> 8);
+                result[1] = (byte) unscaledValue;
+                break;
+            case 3:
+                result[0] = (byte) (unscaledValue >> 16);
+                result[1] = (byte) (unscaledValue >> 8);
+                result[2] = (byte) unscaledValue;
+                break;
+            case 4:
+                result[0] = (byte) (unscaledValue >> 24);
+                result[1] = (byte) (unscaledValue >> 16);
+                result[2] = (byte) (unscaledValue >> 8);
+                result[3] = (byte) unscaledValue;
+                break;
+            case 5:
+                result[0] = (byte) (unscaledValue >> 32);
+                result[1] = (byte) (unscaledValue >> 24);
+                result[2] = (byte) (unscaledValue >> 16);
+                result[3] = (byte) (unscaledValue >> 8);
+                result[4] = (byte) unscaledValue;
+                break;
+            case 6:
+                result[0] = (byte) (unscaledValue >> 40);
+                result[1] = (byte) (unscaledValue >> 32);
+                result[2] = (byte) (unscaledValue >> 24);
+                result[3] = (byte) (unscaledValue >> 16);
+                result[4] = (byte) (unscaledValue >> 8);
+                result[5] = (byte) unscaledValue;
+                break;
+            case 7:
+                result[0] = (byte) (unscaledValue >> 48);
+                result[1] = (byte) (unscaledValue >> 40);
+                result[2] = (byte) (unscaledValue >> 32);
+                result[3] = (byte) (unscaledValue >> 24);
+                result[4] = (byte) (unscaledValue >> 16);
+                result[5] = (byte) (unscaledValue >> 8);
+                result[6] = (byte) unscaledValue;
+                break;
+            case 8:
+                result[0] = (byte) (unscaledValue >> 56);
```

Comment:
in https://github.com/trinodb/trino/pull/12658#discussion_r893526883

i meant something more like


```java
            case 8:
                result[0] = (byte) (unscaledValue >> 56);
                 // fall through
            case 7:
                result[0] = (byte) (unscaledValue >> 48);
                 // fall through
            case 6:
                result[0] = (byte) (unscaledValue >> 40);
```

(see the code linked there)